projects/extension/src/background/ConnectionManager.ts

```diff
@@ -153,6 +156,23 @@ export class ConnectionManager extends (EventEmitter as {
     return client.terminate()
   }
 
+  flushDatabases(): void {
+    this.#chainConnections.forEach(async (ch) => {
+      if (Array.from(wellKnownChains.keys()).includes(ch.chainName)) {
+        const db = await ch.chain?.databaseContent(
+          /**
+           * QUOTA_BYTES: 5242880 is the maximum amount (in bytes) of data that can be
+           * stored in local storage, as measured by the JSON stringification of every
+           * value plus every key's length.
+           * (see https://developer.chrome.com/docs/extensions/reference/storage/#property-local)
+           */
+          chrome.storage.local.QUOTA_BYTES / wellKnownChains.size,
+        )
+        chrome.storage.local.set({ [ch.chainName]: db })
```

Comment:
If we were doing this from the outside, we would be able to use the key of the well-known-chain as the source of truth, rather than using untrusted input.